### ISSUE TITLE
[HOLD for payment 2023-02-23] [$2000] bottom home bar on iOS doesn’t have a transparent background on the on transactions detail page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14378]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open app
2. Tap IOU badge at the top right to view the details page

## Expected Result:
bottom home bar should have a transparent background

## Actual Result:
Bottom home bar doesn’t have a transparent background

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] iOS / native

**Version Number:** 1.2.55-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

<img src="https://user-images.githubusercontent.com/43996225/213018323-a1f73fa9-754b-4a79-b3c0-7db9946b90f9.PNG" width="40%" height="40%">

<img src="https://user-images.githubusercontent.com/43996225/213018352-dd460e42-b4ed-4690-91a7-7ff4a2302eb9.png" width="40%" height="40%">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @fedirjh 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673987346303569

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ddb9b912207d141d</li>
        <li>Upwork Job ID: 1615474343103709184</li>
        <li>Last Price Increase: 2023-02-28</li>
    </ul>
</details>

**Comment by fedirjh** [comment]
### Proposal

This was missed during this PR https://github.com/Expensify/App/pull/13580 , we have to disable `includeSafeAreaPaddingBottom`

```diff
diff --git a/src/pages/iou/IOUDetailsModal.js b/src/pages/iou/IOUDetailsModal.js
index ab600cfb5c..bd099acbdc 100644
--- a/src/pages/iou/IOUDetailsModal.js
+++ b/src/pages/iou/IOUDetailsModal.js
@@ -155,7 +155,7 @@ class IOUDetailsModal extends Component {
         const iouReportStateNum = lodashGet(this.props.iouReport, 'stateNum');
         const hasOutstandingIOU = lodashGet(this.props.iouReport, 'hasOutstandingIOU');
         return (
-            <ScreenWrapper>
+            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
                 <FullPageNotFoundView
                     shouldShow={_.isEmpty(this.props.iouReport)}
                     subtitleKey="notFound.iouReportNotFound"

```

**Comment by situchan** [comment]
### **Proposal:**

[Previous proposal](https://github.com/Expensify/App/issues/14378#issuecomment-1386167575) didn't handle case when settlement button shows at the bottom

In [IOUDetailsModal](https://github.com/Expensify/App/blob/main/src/pages/iou/IOUDetailsModal.js):
```diff
        const hasOutstandingIOU = lodashGet(this.props.iouReport, 'hasOutstandingIOU');
+       const hasFixedFooter = hasOutstandingIOU && this.props.iouReport.managerEmail === sessionEmail;
        return (
-           <ScreenWrapper>
+           <ScreenWrapper includeSafeAreaPaddingBottom={hasFixedFooter}>
+               {({safeAreaPaddingBottomStyle}) => (
                    <FullPageNotFoundView
                        shouldShow={_.isEmpty(this.props.iouReport)}
                        subtitleKey="notFound.iouReportNotFound"
                        onBackButtonPress={Navigation.dismissModal}
                    >
                        <HeaderWithCloseButton
                            title={this.props.translate('common.details')}
                            onCloseButtonPress={Navigation.dismissModal}
                        />
                        {this.props.iou.loading ? <ActivityIndicator color={themeColors.text} /> : (
                            <View style={[styles.flex1, styles.justifyContentBetween]}>
-                               <ScrollView contentContainerStyle={styles.iouDetailsContainer}>
+                               <ScrollView contentContainerStyle={[styles.iouDetailsContainer, safeAreaPaddingBottomStyle]}>
                                    <IOUPreview
                                        chatReportID={this.props.route.params.chatReportID}
                                        iouReportID={this.props.route.params.iouReportID}
                                        pendingAction={pendingAction}
                                        shouldHidePayButton
                                    />
                                    <IOUTransactions
                                        chatReportID={this.props.route.params.chatReportID}
                                        iouReportID={this.props.route.params.iouReportID}
                                        isIOUSettled={iouReportStateNum === CONST.REPORT.STATE_NUM.SUBMITTED}
                                        userEmail={sessionEmail}
                                    />
                                </ScrollView>
-                               {(hasOutstandingIOU && this.props.iouReport.managerEmail === sessionEmail && (
+                               {(hasFixedFooter && (
                                <FixedFooter>
                                    <SettlementButton
                                        onPress={paymentMethodType => this.payMoneyRequest(paymentMethodType)}
                                        shouldShowPaypal={Boolean(lodashGet(this.props, 'iouReport.submitterPayPalMeAddress'))}
                                        currency={lodashGet(this.props, 'iouReport.currency')}
                                        enablePaymentsRoute={ROUTES.IOU_DETAILS_ENABLE_PAYMENTS}
                                        addBankAccountRoute={ROUTES.IOU_DETAILS_ADD_BANK_ACCOUNT}
                                        addDebitCardRoute={ROUTES.IOU_DETAILS_ADD_DEBIT_CARD}
                                        chatReportID={this.props.route.params.chatReportID}
                                    />
                                </FixedFooter>
                                ))}
                            </View>
                        )}
                    </FullPageNotFoundView>
+               )}
            </ScreenWrapper>
        );
```

We should also find and fix any other pages where bottom padding is needed in ScrollView or not needed in FixedFooter.

**Comment by Santhosh-Sellavel** [comment]
@situchan Can you share screenshots of how it looks on all platforms after the fix thanks!

**Comment by Santhosh-Sellavel** [comment]
@joelbettner bump!

**Comment by Santhosh-Sellavel** [comment]
Thanks @fedirjh that makes sense!

**Comment by muttmuure** [comment]
Not overdue, being actively worked on

**Comment by joelbettner** [comment]
@Santhosh-Sellavel, @situchan your proposal looks good to me.

> We should also find and fix any other pages where bottom padding is needed in ScrollView or not needed in FixedFooter.

Are you planning on doing that in the PR for this? Any suggestions for a systematic way we can find those pages?

**Comment by situchan** [comment]
> Are you planning on doing that in the PR for this?

yes


> Any suggestions for a systematic way we can find those pages?

I think no automatic way for this. Should find manually all the usages.
I think this issue is continuation of https://github.com/Expensify/App/pull/13580 where they missed these kinds of issues.

**Comment by Santhosh-Sellavel** [comment]
@situchan Can you test and list down the pages that need to address this issue?

**Comment by situchan** [comment]
<details>
<summary>IOU Request/Send</summary>

Before:
![IOUSend](https://user-images.githubusercontent.com/108292595/214531133-bc13e17f-98ec-4269-b1b9-ee45f20e821d.png)
After:
![IOUSend2](https://user-images.githubusercontent.com/108292595/214531140-f94a8f86-55bc-4a0b-99f9-8ac12255e95a.png)


</details>

<details>
<summary>IOU Split</summary>

Before:
![IOUSplit](https://user-images.githubusercontent.com/108292595/214531191-613ed1b3-a8f1-4cb6-baa3-aa303e4bee2d.png)
After:
![IOUSplit2](https://user-images.githubusercontent.com/108292595/214531201-0f7ac583-5104-48ca-b3c3-b712f6b0db39.png)


</details>

<details>
<summary>Manage members</summary>

Before:
![Manage members](https://user-images.githubusercontent.com/108292595/214531278-c11b69d5-3237-404c-9738-1076f1731340.png)
After:
![Manage members2](https://user-images.githubusercontent.com/108292595/214531286-32d56e05-231c-4801-a6a0-1618247639f7.png)


</details>

<details>
<summary>Members (no transparent background at the bottom)</summary>

Before:
![Members](https://user-images.githubusercontent.com/108292595/214531332-4a28b772-aadc-4210-b416-96c809191387.png)
After:
![Members1](https://user-images.githubusercontent.com/108292595/214531347-0fc9fb98-d850-4c15-8cf7-cd8a9a7e6f21.png)
![Members2](https://user-images.githubusercontent.com/108292595/214531350-48a60698-0449-4ad9-b026-424575ea3f21.png)


</details>

<details>
<summary>New chat</summary>

Before:
![New chat](https://user-images.githubusercontent.com/108292595/214531410-7ee46d14-e04b-44cc-857b-6e007b5d3dfb.png)
After:
![New chat2](https://user-images.githubusercontent.com/108292595/214531423-5f213a54-e7db-4636-af8d-95a52ad04c00.png)


</details>

<details>
<summary>New group</summary>

Before:
![New group](https://user-images.githubusercontent.com/108292595/214531474-30c987e6-d60e-4ac7-a9bb-0f119d40b262.png)
After:
![New group2](https://user-images.githubusercontent.com/108292595/214531481-78d663bf-b5a8-441e-ac27-da49c4296567.png)


</details>

<details>
<summary>Pronouns</summary>

Before:
![Pronouns](https://user-images.githubusercontent.com/108292595/214531522-6e5936d5-0c6b-45c7-9a45-3ba434dda583.png)
After:
![Pronouns2](https://user-images.githubusercontent.com/108292595/214531527-9ffb3576-cc47-4e96-b483-ce2adb0e3aa4.png)


</details>

<details>
<summary>Search</summary>

Before:
![Search](https://user-images.githubusercontent.com/108292595/214531702-1532e11c-48c6-4794-a610-27362a01f1be.png)
After:
![Search2](https://user-images.githubusercontent.com/108292595/214531711-5e181065-07bd-456a-9ded-14cb3f64a721.png)


</details>

<details>
<summary>Select currency</summary>

Before:
![Select currency](https://user-images.githubusercontent.com/108292595/214531786-fb663a4a-9b0d-4181-82f3-ad2944e6e98b.png)
After:
![Select currency2](https://user-images.githubusercontent.com/108292595/214531797-d1c790c4-d78b-4168-a58e-0c1e81cb3064.png)


</details>

<details>
<summary>Timezone</summary>

Before:
![Timezone](https://user-images.githubusercontent.com/108292595/214531840-bd9ce0ec-bcc7-4f04-bc2d-c670fa7e6357.png)
After:
![Timezone2](https://user-images.githubusercontent.com/108292595/214531847-6f0c1265-65dd-4158-a231-528fd4e116c5.png)


</details>

<details>
<summary>About</summary>

Before:
![About](https://user-images.githubusercontent.com/108292595/214530850-4d181bf5-f04a-4b47-ae59-00eb2b7b0f09.png)
After:
![About2](https://user-images.githubusercontent.com/108292595/214530890-f212dd89-1804-4693-b1a1-42f0ad29a852.png)


</details>

- Report Participants page has exactly same issue as IOU Detail page in this GH.
Other pages just need bottom padding in scroll view to avoid overlapping with safe area.

- Also, I found typo in Profile page. There's no `includeSafeAreaPadding` prop in `ScreenWrapper`
```diff
-           <ScreenWrapper includeSafeAreaPadding={false}>
+           <ScreenWrapper includeSafeAreaPaddingBottom={false}>
```

- There are still many pages with ScrollView inside ScreenWrapper but I don't think we need to handle all of them unless content overflows.
i.e. SecuritySettingsPage is scrollable but has only Change password and Close account menu items which don't take full height.

- This is minor but when offline, we don't need to pass `safeAreaPaddingBottomStyle` to children in `ScreenWrapper` since this is already applied to `OfflineIndicator`. (Diff screenshots attached below)
https://github.com/Expensify/App/blob/a9f1892c76ec6e07743660727c144b0069fb59c5/src/components/ScreenWrapper/index.js#L79-L82
```diff
                                        ? this.props.children({
                                            insets,
-                                           safeAreaPaddingBottomStyle,
+                                           safeAreaPaddingBottomStyle: this.props.network.isOffline ? {} : safeAreaPaddingBottomStyle,
                                            didScreenTransitionEnd: this.state.didScreenTransitionEnd,
```

<details>
<summary>Diff</summary>

With:
![offline](https://user-images.githubusercontent.com/108292595/214537614-8a137249-ee0a-4624-89a1-07260260fa17.png)

Without:
![offline2](https://user-images.githubusercontent.com/108292595/214537663-c8774fad-55d0-4d90-bb6c-7df996b8e038.png)

</details>

Full code diff is [here](https://github.com/Expensify/App/pull/14545/files)

**Comment by Santhosh-Sellavel** [comment]
@Expensify/design Can you check this https://github.com/Expensify/App/issues/14378#issuecomment-1403388775 confirm is the  expected correct across all the pages?

**Comment by situchan** [comment]
NOTE: above screenshots are when fully scrolled down to the end

**Comment by shawnborton** [comment]
Thanks so much for the detailed screenshots, they look good to me 👍

**Comment by Santhosh-Sellavel** [comment]
@situchan I left a few comments on the diff, please check.

Also few things,
1. Won't it affect any other platforms, please include screenshots of other platforms for one or two pages?
2. Also the solution kind of looks not DRY, can we somehow avoid this?

**Comment by situchan** [comment]
@Santhosh-Sellavel which one do you think is not DRY?

**New Chat page:** (looks same as production on all other platforms)
<details>
<summary>Web</summary>

<img width="981" alt="web" src="https://user-images.githubusercontent.com/108292595/214957493-0475faea-9a9f-4633-8840-519578de3acd.png">


</details>

<details>
<summary>Mobile Web - Safari</summary>

![msafari](https://user-images.githubusercontent.com/108292595/214957452-bd8632af-113d-402c-b6a9-46e6c31b3114.png)

</details>

<details>
<summary>Android</summary>

![android](https://user-images.githubusercontent.com/108292595/214957703-7286cd0a-b878-4fb6-8cc9-255ee2b942c5.jpg)

</details>

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel which one do you think is not DRY?
Never mind!

**Comment by shawnborton** [comment]
Not overdue

**Comment by Santhosh-Sellavel** [comment]
@joelbettner 
Somehow I missed to add a comment @situchan [proposal](https://github.com/Expensify/App/issues/14378#issuecomment-1403388775) looks good.

Do you have any questions?

**Comment by situchan** [comment]
> @situchan please refrain creating a PR before the job is assigned to you, go ahead from the team. Thanks!

@Santhosh-Sellavel Sure, noted. Thanks for suggestion. The purpose of that draft PR was just for reviewers to check code diff and test easily 🙂

**Comment by Santhosh-Sellavel** [comment]
If that's the case create a PR against your personal fork, which should do for proposal review purposes!

**Comment by Santhosh-Sellavel** [comment]
@joelbettner bump seems there is unnecessary doubling after reviewing the proposal!

cc: @muttmuure!

**Comment by joelbettner** [comment]
This looks good to me.

**Comment by situchan** [comment]
@joelbettner can you please assign me to the issue so I can start working on PR?

**Comment by Santhosh-Sellavel** [comment]
@situchan create a new PR, to avoid the confusion with dates!

**Comment by marcaaron** [comment]
I would suggest we align the pricing of this one back to the time when the proposal was accepted: https://github.com/Expensify/App/issues/14378#issuecomment-1387700979

We are have been waiting for the assignment from @joelbettner during the doubling period. Does everyone agree with this logic?

**Comment by Santhosh-Sellavel** [comment]
There is been some additional review posts that and it affects so many pages and this was the final review [comment](https://github.com/Expensify/App/issues/14378#issuecomment-1410635694).  I think we should revert to $2k as I said [here](https://github.com/Expensify/App/issues/14378#issuecomment-1412763712).

**Comment by situchan** [comment]
[PR](https://github.com/Expensify/App/pull/14896) is ready for review

**Comment by Santhosh-Sellavel** [comment]
Thanks for the bump, will continue reviewing tomorrow!

**Comment by situchan** [comment]
any feedback on PR?

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $1000

**Comment by situchan** [comment]
Since this affects many pages across the app I think 2k is fair. And I believe there will be no additional bonus here since PR merged longer than 3 days.

**Comment by Santhosh-Sellavel** [comment]
> Since this affects many pages across the app I think 2k is fair. And I believe there will be no additional bonus here since PR merged longer than 3 days. However, I will follow the team's decision for the final amount.

I agree with this https://github.com/Expensify/App/issues/14378#issuecomment-1419542506.  Requesting to increase, thanks!

cc:  @muttmuure @joelbettner

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.72-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/14896


If no regressions arise, payment will be issued on 2023-02-23. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel / @joelbettner] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel / @joelbettner] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel / @joelbettner] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@situchan] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@muttmuure] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by shawnborton** [comment]
Not overdue

**Comment by Santhosh-Sellavel** [comment]
@joelbettner 

> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  [@Santhosh-Sellavel / @joelbettner] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/13580 Missed here
> * [x]  [@Santhosh-Sellavel / @joelbettner] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/13580#issuecomment-1446707935
> * [x]  [@Santhosh-Sellavel / @joelbettner] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
Adding regression steps should be fine, let me know your thoughts @joelbettner 
> * [ ]  [@situchan] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
> * [ ]  [@muttmuure] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by situchan** [comment]
**Regression Test Proposal**

1. Login with any account that has many IOU transactions with one contact.
2. Click that contact from LHN.
3. Go to IOU detail page by clicking any of IOU transactions on Report screen.
4. Scroll down the content.
5. Verify that bottom content is not cut off by safe area.
6. Verify 5. on all pages across the app.
- New chat
- New group
- Search
- Select currency
- IOU Request/Send
- IOU Split
- IOU Transactions detail (added in test step)
- Workspace
- Manage members
- Workspace Members
- Profile
- Settings
- About
- Pronouns
- Timezone

Do we agree 👍 or 👎

**Comment by Santhosh-Sellavel** [comment]
Define Bug statement otherwise good!

**Comment by situchan** [comment]
> Define Bug statement otherwise good!

What do you mean "Bug statement"?

**Comment by Santhosh-Sellavel** [comment]
@situchan 

Examples from [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md)
<img width="736" alt="Screenshot 2023-02-27 at 10 50 04 PM" src="https://user-images.githubusercontent.com/85645967/221634213-26ba0787-4b6c-4673-8c77-1278543c9c7a.png">

**Comment by Santhosh-Sellavel** [comment]
Bump @muttmuure 

> > Since this affects many pages across the app I think 2k is fair. And I believe there will be no additional bonus here since PR merged longer than 3 days. However, I will follow the team's decision for the final amount.
> 
> I agree with this [#14378 (comment)](https://github.com/Expensify/App/issues/14378#issuecomment-1419542506). Requesting to increase, thanks!
> 
> cc: @muttmuure @joelbettner

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by muttmuure** [comment]
Invited @Santhosh-Sellavel @situchan @fedirjh to the Upwork job

**Comment by muttmuure** [comment]
Everyone has been paid. Good hustle everyone.
